File: python/exercise_303.py
#poderia salvar os resultados num array pra otimizar runtime
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def sumRange(self, left: int, right: int) -> int:
        soma = 0
        for i in range(left,right+1):
            soma += self.nums[i]
        return soma


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: java/arrays/exercise_303.java
class NumArray {
    List<Integer> sums = new ArrayList<>();
    public NumArray(int[] nums) {
        int sum = 0;
        for(int num: nums){
            sum +=num;
            sums.add(sum);
        }
    }
    
    public int sumRange(int left, int right) {
        if(left==0){
            return sums.get(right);
        }
        return(sums.get(right)-sums.get(left-1));
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */